src/math/bigintops.c

```diff
@@ -1184,7 +1204,9 @@ MVMObject * MVM_bigint_rand(MVMThreadContext *tc, MVMObject *type, MVMObject *b)
 
     if (use_small_arithmetic) {
         if (MP_GEN_RANDOM_MAX >= abs(smallint_max)) {
-            mp_digit result_int = rand() % smallint_max;
+            mp_digit result_int;
+            MVM_getrandom(tc, &result_int, MP_GEN_RANDOM_MAX);
```

Comment:
I guess you want to check the return value ... (just in case something's really broken)